pkg/storage/s3/s3.go

```diff
@@ -75,3 +75,10 @@ func New(s3Conf *config.S3Config, timeout time.Duration, options ...func(*aws.Co
 		timeout:  timeout,
 	}, nil
 }
+
+func endpointFrom(credentialsEndpoint string, relativeURI string) string {
+	if relativeURI == "" {
+		return credentialsEndpoint
+	}
+	return credentialsEndpoint + "$" + relativeURI
```

Comment:
I was only using `$` in my previous comments to indicate this is an environment variable, if you're using `envconfig` to resolve `AWS_CONTAINER_CREDENTIALS_RELATIVE_URI`, then you won't need this `$`

Reply:
makes sense! sorry about the mistake. will put in a new revision.